src/System.IO.MemoryMappedFiles/src/Microsoft/Win32/SafeMemoryMappedFileHandle.Unix.cs

```diff
@@ -2,24 +2,68 @@
 using System.Runtime.InteropServices;
 using System.Security;
+using System.Threading;
 
 namespace Microsoft.Win32.SafeHandles
 {
     public sealed partial class SafeMemoryMappedFileHandle : SafeHandle
     {
-        private const int DefaultInvalidHandleValue = -1; // TODO: Implement this
+        /// <summary>Counter used to produce a unique handle value.</summary>
+        private static long s_counter = 0;
 
-        protected override bool ReleaseHandle()
+        /// <summary>The underlying SafeFileHandle.  May be null to use anonymous backing storage.</summary>
+        internal readonly SafeFileHandle _fileHandle;
+
+        /// <summary>The inheritability of the memory-mapped file.</summary>
+        internal readonly HandleInheritability _inheritability;
+
+        /// <summary>The access to the memory-mapped file.</summary>
+        internal readonly MemoryMappedFileAccess _access;
+
+        /// <summary>The options for the memory-mapped file.</summary>
+        internal readonly MemoryMappedFileOptions _options;
+
+        /// <summary>The capacity of the memory-mapped file.</summary>
+        internal readonly Int64 _capacity;
+
+        /// <summary>Initializes the memory-mapped file handle.</summary>
+        /// <param name="fileHandle">The underlying file handle; this may be null in the case of a page-file backed memory-mapped file.</param>
+        /// <param name="inheritability">The inheritability of the memory-mapped file.</param>
+        /// <param name="access">The access for the memory-mapped file.</param>
+        /// <param name="options">The options for the memory-mapped file.</param>
+        /// <param name="capacity">The capacity of the memory-mapped file.</param>
+        internal SafeMemoryMappedFileHandle(
+            SafeFileHandle fileHandle, HandleInheritability inheritability,
+            MemoryMappedFileAccess access, MemoryMappedFileOptions options,
+            long capacity)
+            : base(new IntPtr(-1), ownsHandle: true)
+        {
+            // Store the arguments.  We'll actually open the map when the view is created.
+            _fileHandle = fileHandle;
+            _inheritability = inheritability;
+            _access = access;
+            _options = options;
+            _capacity = capacity;
+
+            // Fake a unique Int32 handle value > 0.
+            int nextHandleValue = (int)((Interlocked.Increment(ref s_counter) % (int.MaxValue - 1)) + 1);
+            SetHandle(new IntPtr(nextHandleValue));
+        }
+
+        protected override unsafe bool ReleaseHandle()
         {
-            throw NotImplemented.ByDesign; // TODO: Implement this
+            // The actual handle we're holding in the SafeHandle is fake, so nothing to do.
+            return true;
         }
 
         public override bool IsInvalid
         {
             [SecurityCritical]
-            get { throw NotImplemented.ByDesign; } // TODO: Implement this
+            get { return (long)handle <= 0; }
```

Comment:
Since IntPtr is a long on 64-bit, I just wanted to account for the possibility that there could be some garbage in here.  It's unlikely, I know, but I don't think it hurts anything to use the larger data type.